airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/DefaultJobPersistence.java

```diff
@@ -353,6 +362,39 @@ public List<Job> listJobs(final Set<ConfigType> configTypes, final String config
         jobSelectAndJoin(jobsSubquery) + ORDER_BY_JOB_TIME_ATTEMPT_TIME)));
   }
 
+  @Override
+  public List<Job> listJobsIncludingId(final Set<ConfigType> configTypes, final String connectionId, final long targetJobId, final int pagesize)
+      throws IOException {
+    final Optional<OffsetDateTime> targetJobCreatedAt = jobDatabase.query(ctx -> ctx.select(JOBS.CREATED_AT).from(JOBS)
+        .where(JOBS.CONFIG_TYPE.cast(String.class).in(Sqls.toSqlNames(configTypes)))
+        .and(JOBS.SCOPE.eq(connectionId))
+        .and(JOBS.ID.eq(targetJobId))
+        .stream()
+        .findFirst()
+        .map(record -> record.get(JOBS.CREATED_AT, OffsetDateTime.class)));
+
+    if (targetJobCreatedAt.isEmpty()) {
+      return List.of();
+    }
+
+    final int countIncludingTargetJob = jobDatabase.query(ctx -> ctx.selectCount().from(JOBS)
+        .where(JOBS.CONFIG_TYPE.cast(String.class).in(Sqls.toSqlNames(configTypes)))
+        .and(JOBS.SCOPE.eq(connectionId))
+        .and(JOBS.CREATED_AT.greaterOrEqual(targetJobCreatedAt.get()))
+        .fetchOne().into(int.class));
+
+    // always want to return at least `pagesize` number of jobs
+    if (countIncludingTargetJob < pagesize) {
+      return listJobs(configTypes, connectionId, pagesize, 0);
+    }
+
+    // list of jobs up to target job is larger than pagesize, so return that list
+    final String jobsSubquery = "(SELECT * FROM jobs WHERE CAST(config_type AS VARCHAR) in " + Sqls.toSqlInFragment(configTypes)
```

Comment:
Should we convert that to jooq as well?

Reply:
I think that one is a bit harder to convert to jooq, because it is using the `jobSelectAndJoin()` method which takes in the subquery and inserts that into the SELECT statement string that it builds. However, I may be able to build the query with jooq and then call something like `toSql()` to convert it into a string that can be passed to that method. I'll play around with that for a bit but no guarantees on if I can get it working properly

Reply:
I was able to convert these to jooq using `getSql()` and `DSL.inline()`, so I have pushed that up to this PR as well! Thanks for the suggestion, I think it looks a lot cleaner now